crates/accelerate/src/unitary_synthesis.rs

```diff
@@ -0,0 +1,1093 @@
+            op: PackedOperation::from_standard(gate_node),
+            qubits: out_dag.qargs_interner.insert(&mapped_qargs),
+            clbits: out_dag.cargs_interner.get_default(),
+            params: new_params,
+            extra_attrs: ExtraInstructionAttributes::new(None, None, None, None),
+            #[cfg(feature = "cache_pygates")]
+            py_op: OnceCell::new(),
+        };
+        instructions.push(instruction);
+    }
+    out_dag.extend(py, instructions.into_iter())?;
+    out_dag.add_global_phase(py, &Param::Float(sequence.gate_sequence.global_phase))?;
+    Ok(())
+}
+
+fn synth_error(
+    py: Python<'_>,
+    synth_circuit: impl Iterator<
+        Item = (
+            String,
+            Option<SmallVec<[qiskit_circuit::operations::Param; 3]>>,
+            SmallVec<[PhysicalQubit; 2]>,
+        ),
+    >,
+    target: &Target,
+) -> f64 {
+    let mut gate_fidelities = Vec::new();
+    let mut score_instruction = |inst_name: &str,
+                                 inst_params: &Option<SmallVec<[Param; 3]>>,
+                                 inst_qubits: &SmallVec<[PhysicalQubit; 2]>|
+     -> PyResult<()> {
+        match target.operation_names_for_qargs(Some(inst_qubits)) {
+            Ok(names) => {
+                for name in names {
+                    let target_op = target.operation_from_name(name).unwrap();
+                    let are_params_close = if let Some(params) = inst_params {
+                            params.iter().zip(target_op.params.iter()).all(|(p1, p2)| {
+                                p1.is_close(py, p2, 1e-10)
+                                    .expect("Unexpected parameter expression error.")
+                            })
+                    } else {
+                        false
+                    };
+                    let is_parametrized = target_op
+                        .params
+                        .iter()
+                        .any(|param| matches!(param, Param::ParameterExpression(_)));
+                    if target_op.operation.name() == inst_name
+                        && (is_parametrized || are_params_close)
+                    {
+                        match target[name].get(Some(inst_qubits)) {
+                            Some(Some(props)) => gate_fidelities.push(
+                                1.0 - props.error.unwrap_or(0.0)
+                            ),
+                            _ => gate_fidelities.push(1.0),
+                        }
+                        break;
+                    }
+                }
+                Ok(())
+            }
+            Err(_) => {
+                Err(QiskitError::new_err(
+                    format!("Encountered a bad synthesis. Target has no instruction {inst_name:?} on qubits {inst_qubits:?}.")
+                ))
+            }
+        }
+    };
+
+    for (inst_name, inst_params, inst_qubits) in synth_circuit {
+        let _ = score_instruction(&inst_name, &inst_params, &inst_qubits);
```

Comment:
Thanks for catching this. Indeed the error wasn't handled, but I also don't think it's even necessary to contemplate this case because if the target didn't have any operations in the specified qargs, an error would have been raised already in line 570. So I'd say it's safe to remove that arm and turn the match into an if let, done in [78848ec](https://github.com/Qiskit/qiskit/pull/13141/commits/78848ecd70bc226247cdf8830a5131670c0078cc).